Providers/Profile.swift

```diff
@@ -1082,8 +1082,29 @@ open class BrowserProfile: Profile {
                                   account: FirefoxAccount,
                                   statsSession: SyncOperationStatsSession, why: SyncReason) -> Deferred<Maybe<[(EngineIdentifier, SyncStatus)]>> {
             log.info("Syncing \(synchronizers.map { $0.0 })")
-            let authState = account.syncAuthState
+            var authState = account.syncAuthState
             let delegate = self.profile.getSyncDelegate()
+
+            var enginesEnablements: [String: Bool] = [:]
+            // We just created the account, the user went through the Choose What to Sync screen on FxA.
+            if let declined = account.declinedEngines {
+                declined.forEach { enginesEnablements[$0] = false }
+                account.declinedEngines = nil
+                profile.flushAccount()
+            } else {
+                // Bundle in authState the engines the user activated/disabled.
+                let togglableEngines = ["bookmarks", "history", "tabs", "passwords"]
```

Comment:
Make this a constant, put it next to, and relate it to `SyncStateMachine.LocalEngines` (which you might want to move somewhere else) so they don't drift.